libyul/optimiser/ReasoningBasedSimplifier.cpp

```diff
@@ -56,14 +56,8 @@ std::optional<string> ReasoningBasedSimplifier::invalidInCurrentEnvironment()
 
 void ReasoningBasedSimplifier::operator()(VariableDeclaration& _varDecl)
 {
-	if (_varDecl.variables.size() != 1 || !_varDecl.value)
-		return;
-	YulString varName = _varDecl.variables.front().name;
-	if (!m_ssaVariables.count(varName))
-		return;
-	bool const inserted = m_variables.insert({varName, m_solver->newVariable("yul_" + varName.str(), defaultSort())}).second;
-	yulAssert(inserted, "");
-	m_solver->addAssertion(m_variables.at(varName) == encodeExpression(*_varDecl.value));
+	ASTModifier::operator()(_varDecl);
```

Comment:
Visit to `_varDecl.value` was missing before.

Reply:
Why does it need the visit?

Reply:
Actually, it doesn't need it for ReasoningBasedSimplifier. Should I remove it?

Reply:
Yes, please remove it.